src/state.rs

```diff
@@ -369,129 +448,90 @@ impl Persistent {
         Ok(())
     }
 
-    pub fn remaining_tries(&self, password: Password) -> u8 {
-        match password {
-            Password::Pw1 => Self::MAX_RETRIES.saturating_sub(self.user_pin_tries),
-            Password::Pw3 => Self::MAX_RETRIES.saturating_sub(self.admin_pin_tries),
-            Password::ResetCode => Self::MAX_RETRIES.saturating_sub(self.reset_code_tries),
-        }
+    pub fn remaining_tries<T: trussed::Client + AuthClient>(
+        &self,
+        client: &mut T,
+        password: Password,
+    ) -> u8 {
+        try_syscall!(client.pin_retries(password))
+            .map(|r| r.retries.unwrap_or_default())
+            .unwrap_or(Self::MAX_RETRIES)
```

Comment:
Return 0 instead?  If `try_syscall` is `Err`, the PIN is not set for some reason, so it cannot be verified.

Reply:
This makes me realize that we were always returning `3` for the PUK, when `0` makes more sense, which this would fix.